File: leetcode/editor/en/[1658]Minimum Operations to Reduce X to Zero.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @param {number} x
 * @return {number}
 */
var minOperations = function (nums, x) {
    let total = nums.reduce((a, c) => a + c);
    const n = nums.length;
    const target = total - x;
    let max = -1;
    let left = 0;
    let current = 0;
    for (let right = 0; right < n; right++) {
        current += nums[right];

        while (current > target && left <= right) {
            current -= nums[left];
            left++;
        }
        if (current === target) {
            max = Math.max(max, right - left + 1);
        }
    }

    return max === -1 ? -1 : n - max;
};
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[1658]Minimum Operations to Reduce X to Zero.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int minOperations(int[] nums, int x) {
        int total = Arrays.stream(nums).sum();
        int n = nums.length;
        int target = total - x;
        int max = -1;
        int left = 0;
        int current = 0;

        for (int right = 0; right < n; right++) {
            current += nums[right];

            while ((current > target) && left <= right) {
                current -= nums[left];
                left++;
            }

            if (current == target) {
                max = Math.max(max, right - left + 1);
            }
        }

        return max == -1 ? -1 : n - max;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
